File: lib/domain/farm/dtos/start_farm_weather_data_fetch_input.rb
# frozen_string_literal: true

module Domain
  module Farm
    module Dtos
      class StartFarmWeatherDataFetchInput
        attr_reader :farm_id, :as_of

        def initialize(farm_id:, as_of:)
          @farm_id = farm_id
          @as_of = as_of
        end
      end
    end
  end
end

File: crates/agrr-domain/src/farm/dtos/start_farm_weather_data_fetch_input.rs
use time::Date;

/// Ruby: `Domain::Farm::Dtos::StartFarmWeatherDataFetchInput`
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub struct StartFarmWeatherDataFetchInput {
    pub farm_id: i64,
    pub as_of: Date,
}
